src/lib/logical_query_plan/abstract_lqp_node.hpp

```diff
@@ -172,9 +172,23 @@ class AbstractLQPNode : public std::enable_shared_from_this<AbstractLQPNode> {
 
   /**
    * @return The functional dependencies valid for this node. See functional_dependency.hpp for documentation.
-   *         By default, functional dependencies from both sides are forwarded. Nodes may override this behavior.
+   *         They are collected as follows:
+   *          (1) FDs are derived from the current node's unique constraints.
+   *          (2) FDs are provided by previous nodes by passing them up the LQP tree.
    */
-  virtual std::vector<FunctionalDependency> functional_dependencies() const;
+  std::vector<FunctionalDependency> functional_dependencies() const;
+
+  /**
+   * Helper function that passes FDs to upper nodes. By default, it passes FDs from the left input node.
+   * Some nodes may override this function to
+   *  - add additional FDs or
+   *  - to specify FD passing in case of two input nodes.
+   *
+   *  Please note: Do not pass FDs that can be generated from unique constraints because functional_dependencies()
+   *               already takes care of that. However, whenever unique constraints are discarded, it makes sense to
+   *               do so since FDs cannot be generated later on anymore.
```

Comment:
Could you rephrase the second sentence?

Reply:
Changed to
```
*  Please note: Do not pass FDs that can be generated from unique constraints because functional_dependencies()
*               already takes care of that.
*               However, there is an exception: Some nodes, such as JoinNode, discard unique constraints.
*               Consequently, we lose FDs that can no longer be derived in upper nodes. To preserve them, we might
*               want to generate FDs from unique constraints prematurely to manually pass them up the LQP tree.
```

Reply:
Yepp. :)